lighthouse-core/audits/resource-budget.js

```diff
@@ -0,0 +1,182 @@
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      requiredArtifacts: ['devtoolsLogs', 'URL'],
+    };
+  }
+
+  /**
+   * @param {LH.Budget | undefined} budget
+   * @return {LH.Audit.Details.Table['headings']}
+   */
+  static tableHeadings(budget) {
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headers = [
+      {key: 'label', itemType: 'text', text: 'Resource Type'},
+      {key: 'requestCount', itemType: 'numeric', text: 'Requests'},
+      {key: 'size', itemType: 'bytes', text: 'Transfer Size'},
+    ];
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const budgetHeaders = [
+      {key: 'countOverBudget', itemType: 'text', text: ''},
+      {key: 'sizeOverBudget', itemType: 'bytes', text: 'Over Budget'},
+    ];
+    return budget ? headers.concat(budgetHeaders) : headers;
+  }
+
+  /**
+   * @param {LH.Budget.ResourceType} resourceType
+   * @return {string}
+   */
+  static rowLabel(resourceType) {
+    /** @type {Record<LH.Budget.ResourceType,string>} */
+    const strMappings = {
+      'total': str_(i18n.UIStrings.totalResourceType),
+      'document': str_(i18n.UIStrings.documentResourceType),
+      'script': str_(i18n.UIStrings.scriptResourceType),
+      'stylesheet': str_(i18n.UIStrings.stylesheetResourceType),
+      'image': str_(i18n.UIStrings.imageResourceType),
+      'media': str_(i18n.UIStrings.mediaResourceType),
+      'font': str_(i18n.UIStrings.fontResourceType),
+      'other': str_(i18n.UIStrings.otherResourceType),
+      'third-party': str_(i18n.UIStrings.thirdPartyResourceType),
+    };
+    return strMappings[resourceType];
+  }
+
+  /**
+   * @param {Record<LH.Budget.ResourceType,ResourceEntry>} summary
+   * @return {Array<{resourceType: LH.Budget.ResourceType, label: string, requestCount: number, size: number}>}
+   */
+  static defaultTable(summary) {
+    const types = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(summary));
+    const rows = types.map(type => {
+      return {
+        resourceType: type,
+        label: this.rowLabel(type),
+        requestCount: summary[type].count,
+        size: summary[type].size,
+      };
+    });
+    // Force third-party to be last, descending by size otherwise
+    const thirdPartyRow = rows.find(r => r.resourceType === 'third-party') || [];
+    const otherRows = rows.filter(r => r.resourceType !== 'third-party')
+      .sort((a, b) => {
+        return b.size - a.size;
+      });
+    return otherRows.concat(thirdPartyRow);
+  }
+
+  /**
+   * @param {LH.Budget} budget
+   * @param {Record<LH.Budget.ResourceType,ResourceEntry>} summary
+   * @return {Array<{resourceType: LH.Budget.ResourceType, label: string, requestCount: number, size: number, sizeOverBudget: number | undefined, countOverBudget: string | undefined}>}
+   */
+  static budgetTable(budget, summary) {
+    const resourceTypes = /** @type {Array<LH.Budget.ResourceType>} */ (Object.keys(summary));
+    return resourceTypes.map((type) => {
```

Comment:
```suggestion
    return resourceTypes.map(resourceType => {
```
(and can delete the line below this)